webui/react/src/contexts/Commands.tsx

```diff
@@ -1,30 +1,104 @@
+import { useCallback } from 'react';
+
 import { generateContext } from 'contexts';
 import { RestApiState } from 'hooks/useRestApi';
-import { Command } from 'types';
+import { CommandTask } from 'types';
 import { clone } from 'utils/data';
 
+import { getCommands, getNotebooks, getShells, getTensorboards } from '../services/api';
+
 const initialState = {
   errorCount: 0,
   hasLoaded: false,
   isLoading: false,
 };
 
-export const Commands = generateContext<RestApiState<Command[]>>({
+export const Commands = generateContext<RestApiState<CommandTask[]>>({
   initialState: clone(initialState),
   name: 'Commands',
 });
 
-export const Notebooks = generateContext<RestApiState<Command[]>>({
+export const useFetchCommands = (canceler: AbortController): () => Promise => {
+  const setCommands = Commands.useActionContext();
+
+  return useCallback(async (): Promise<void> => {
+    const commandsResponse = await getCommands({ signal: canceler.signal });
+    setCommands({
+      type: Commands.ActionType.Set,
+      value: {
+        data: commandsResponse,
+        errorCount: 0,
+        hasLoaded: true,
+        isLoading: false,
+      },
```

Comment:
question: are we using all the state this is providing anywhere? if not I'd be for simplifying it. or maybe not make this change if we don't need to? 

same for other command types below.

Reply:
Yes, this data structure is still used in:
- webui/react/src/pages/TaskList.tsx
- webui/react/src/pages/Dashboard.tsx

Reply:
yeah, let's not make the pr bigger if we don't need to.
I might be missing something: where is `isLoading` getting set to true?

Reply:
`isLoading` and `errorCount` are (now) never used. What's used is hasLoaded.

Reply:
Since this has to wait to land I can get this right now?

Can simply change TS interface from `RestApiState` to `ApiState` and this will cause those object to have just `isLoading` (to use instead of `hasLoaded`) and `data`.

What do you think?

Reply:
hmm IIRC we were talking about cleaning up the `ApiState` and `RestApiState` types and unifying them. which would also get easier once all apis are migrated to the new API. I'd suggest just leaving as is and keeping the changes smaller here so we also get a chance to talk about it with the team